### PR TITLE
Relationship precedence over attributes 

### PR DESCRIPTION
**Pull request by seanlally**
Potential fix for updaterId issue with prospects table (outreach not going to fix on their end)

Issue: https://github.com/singer-io/tap-outreach/issues/20
Code referenced: https://github.com/Pathlight/tap-outreach/pull/3

### CODE REVIEW

**Comment by asharmaupland** [comment]
Any Update On this ?

**Comment by RushiT0122** [comment]
@nitingaikwad1 please review this PR.

**Comment by sgandhi1311** [comment]
Hello Sean,
I see you are getting the authentication error message - `Error testing Outreach authentication`
Can you please re-auth your outreach integration on the stitch platform? This will refresh the credentials (refresh token / access token). 
Once the re-auth is done I can proactively look into the _updaterId issue_ and get it fixed ASAP.

**Comment by seanlally** [comment]
We are working on resolving our authentication issues. But I am unsure why we need to fix our own authentication issue before the updaterId issue could be fixed. It might  be a couple days before we get our auth issue resolved.

**Comment by seanlally** [comment]
I have new credentials to use, but i cannot update them in stitch because the link to outreach sign on is broken and gives this error: "Client authentication failed due to unknown client, no client authentication included, or unsupported authentication method." It occurs on a new integration being made too. Hopefully this updaterId issue can be resolved without the authorization being fixed.

**Comment by seanlally** [comment]
Stitch mentioned that the authentication problem i'm having is another issue with this integration and not a problem on their end. We can't fix our authentication until that problem is found. it fails for the reason posted previously with both a new and exisiting outreach stitch integration.

**Comment by asharmaupland** [comment]
@seanlally  Any update?

**Comment by seanlally** [comment]
The re-authentication of outreach integrations on Stitch is still broken. There is nothing for me to do.

**Comment by asharmaupland** [comment]
> sgandhi1311
Hi
 Re-auth is fixed. Can you please look into the updaterId issue?

File: tap_outreach/sync.py [hunk 1]
```diff
@@ -227,9 +227,11 @@ def process_records(stream, mdata, max_modified, records, filter_field, fks):
...
                                 'null or `id` field expected for `data` relationship')
 
-                        if fk_field_name in record_flat:
-                            raise Exception(
-                                '`{}` exists as both an attribute and generated relationship name'.format(fk_field_name))
+                        # January 2023: Let relationship value of record take precedence over attribute value (will take precedence at line 214)
+                        # (https://github.com/singer-io/tap-outreach/issues/20)
+                        # if fk_field_name in record_flat:
+                        #     raise Exception(
+                        #         '`{}` exists as both an attribute and generated relationship name'.format(fk_field_name))
```
Comment (sgandhi1311) [inline tap_outreach/sync.py:234]: @seanlally I see the issue specifically for the prospects table as mentioned in the description. Should we consider bypassing the condition for the prospects table?

```suggestion
if stream != "prospects":
     if fk_field_name in record_flat:
          raise Exception(
                                '`{}` exists as both an attribute and generated relationship
                                name'.format(fk_field_name))
```

**Comment by dlachasse** [comment]
Hey @seanlally, thanks for the fix here! I've just traced the exceptions happening in our Outreach extraction that led me to this PR. Any ETA on when this will be released?

**Review by sgandhi1311 — APPROVED** [review]

**Comment by sgandhi1311** [comment]
Hello Everyone, I can unblock you all by releasing the change to prod. But I am just worried whether this updaterId issue is specific to the prospects table or do we observe it in other tables as well.

**Comment by asharmaupland** [comment]
@sgandhi1311  The problem is only in the prospects table others are working fine.

**Comment by sgandhi1311** [comment]
Thanks @seanlally for your contribution!!

I needed to fix the syntax error within the code, so I have created a separate branch from the existing PR (in that case we won't miss your valuable contribution).

Current PR - #25

**Comment by sgandhi1311** [comment]
The fix provided by @seanlally #25  is deployed to prod. Please validate whether you are getting the expected results.

**Comment by dlachasse** [comment]
> The fix provided by @seanlally #25 is deployed to prod. Please validate whether you are getting the expected results.

Excellent, it's working for me! Thank you 🙏

**Comment by sgandhi1311** [comment]
Closing the PR as the changes are merged in prod.

**Pull request closed without merging**